genesyscloud/webdeployments_configuration/utils/resource_genesyscloud_webdeployments_configuration_messenger.go

```diff
@@ -69,15 +69,13 @@ func buildAppKnowledge(knowledge []interface{}) *platformclientv2.Knowledge {
 			Id: &knowledgeBaseId,
 		}
 	}
-
 	return ret
 }
 
 func buildMessengerApps(apps []interface{}) *platformclientv2.Messengerapps {
 	if len(apps) < 1 {
 		return nil
 	}
-
 	app := apps[0].(map[string]interface{})
```

Comment:
Same as with `buildAppKnowledge` above, the only item in apps could be nil, so we'll want to check for that with `if app, ok := apps[0].(map[string]interface{})`  

This can happen when all of the fields in the map are optional, so the end user could define messenger apps like this (even though they don't have any good reason to do so):

```hcl
messenger {
  apps {}
}
```

Because there are no fields, it doesn't register as a map and instead one nil item is added to the apps list.